File: Structural - Exercises/Solutions-Leonie/Bridge/protocols/smtp.py
from protocols.protocol_interface import ProtocolInterface

class SMTPProtocol(ProtocolInterface):
    def initialize(self):
        print("SMTP: Initializing email protocol")

    def send_email(self, message: str):
        print(f"SMTP: Sending email: {message}")
        self.provider.process_email(message)

    def shutdown(self):
        print("SMTP: Shutting down protocol")

File: Lesson.03 - Design Patterns - Structural/Bridge/SMTP.java
public class SMTP implements EmailProtocol {
    @Override
    public void initialize() {
        System.out.println("SMTP: Initializing mail transfer protocol");
    }

    @Override
    public void sendEmail(String message) {
        System.out.println("SMTP: Sending email: " + message);
    }

    @Override
    public void shutdown() {
        System.out.println("SMTP: Shutting down protocol");
    }
}
